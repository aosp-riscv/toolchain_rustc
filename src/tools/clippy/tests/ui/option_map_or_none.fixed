// run-rustfix

<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
=======
#![allow(clippy::option_and_then_some)]

>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
fn main() {
    let opt = Some(1);

    // Check `OPTION_MAP_OR_NONE`.
    // Single line case.
    let _ = opt.and_then(|x| Some(x + 1));
    // Multi-line case.
    #[rustfmt::skip]
    let _ = opt.and_then(|x| {
                        Some(x + 1)
                       });
}
