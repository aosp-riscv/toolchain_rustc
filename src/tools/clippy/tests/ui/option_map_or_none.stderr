error: called `map_or(None, f)` on an Option value. This can be done more directly by calling `and_then(f)` instead
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
  --> $DIR/option_map_or_none.rs:8:13
   |
LL |     let _ = opt.map_or(None, |x| Some(x + 1));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try using and_then instead: `opt.and_then(|x| Some(x + 1))`
   |
   = note: `-D clippy::option-map-or-none` implied by `-D warnings`

error: called `map_or(None, f)` on an Option value. This can be done more directly by calling `and_then(f)` instead
  --> $DIR/option_map_or_none.rs:11:13
=======
  --> $DIR/option_map_or_none.rs:10:13
   |
LL |     let _ = opt.map_or(None, |x| Some(x + 1));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try using and_then instead: `opt.and_then(|x| Some(x + 1))`
   |
   = note: `-D clippy::option-map-or-none` implied by `-D warnings`

error: called `map_or(None, f)` on an Option value. This can be done more directly by calling `and_then(f)` instead
  --> $DIR/option_map_or_none.rs:13:13
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
   |
LL |       let _ = opt.map_or(None, |x| {
   |  _____________^
LL | |                         Some(x + 1)
LL | |                        });
   | |_________________________^
help: try using and_then instead
   |
LL |     let _ = opt.and_then(|x| {
LL |                         Some(x + 1)
LL |                        });
   |

error: aborting due to 2 previous errors

